pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -174,31 +174,31 @@ func (c *Controller) GetService(hostname host.Name) (*model.Service, error) {
 			// VIPs or CIDR ranges in the address field
 			return service, nil
 		}
+
 		// This is K8S typically
 		if out == nil {
-			out = service
+			out = service.DeepCopy()
```

Comment:
Always a deep copy?

Reply:
Yes, it doesn't matter much as currently is it only used for gateway service.